src/_posts/2018-08-XX-review-input-streaming-connectors.md

```diff
@@ -0,0 +1,224 @@
+
+Spark offers two approaches to streaming: [Discretized Streaming](https://spark.apache.org/docs/latest/streaming-programming-guide.html) (or DStreams) and [Structured Streaming](https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html). DStreams are a basic abstraction that represents a continuous series of [Resilient Distributed Datasets](https://spark.apache.org/docs/latest/rdd-programming-guide.html) (or RDDs). Structured Streaming was introduced more recently (the alpha release came with Spark 2.1.0) and is based on a [model](https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html#programming-model) where live data is continuously appended to a table structure.
+
+Spark Structured Streaming supports [file sources](https://spark.apache.org/docs/latest/api/java/org/apache/spark/sql/streaming/DataStreamReader.html) (local filesystems and HDFS-compatible systems like Cloud Storage or S3) and [Kafka](https://spark.apache.org/docs/latest/structured-streaming-kafka-integration.html) as streaming inputs. Spark maintains built-in connectors for DStreams aimed at third-party services, such as Kafka or Flume, while other connectors are available through linking external dependencies, as shown in the table below.
+
+Below are the main streaming input connectors for available for Beam and Spark DStreams in Java:
+
+<table>
+  <tr>
+   <td>
+   </td>
+   <td>
+   </td>
+   <td><strong>Apache Beam</strong>
+   </td>
+   <td><strong>Apache Spark DStreams</strong>
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="2" >File Systems
+   </td>
+   <td>Local<br>(Using the <code>file://</code> URI)
+   </td>
+   <td><a href="https://beam.apache.org/documentation/sdks/javadoc/2.5.0/org/apache/beam/sdk/io/TextIO.html">TextIO</a>
+   </td>
+   <td><a href="https://spark.apache.org/docs/latest/api/java/org/apache/spark/streaming/StreamingContext.html#textFileStream-java.lang.String-">textFileStream</a><br>(Spark treats most Unix systems as HDFS-compatible, but the location should be accessible from all nodes)
+   </td>
+  </tr>
+  <tr>
+   <td>HDFS<br>(Using the <code>hdfs://</code> URI)
+   </td>
+   <td><a href="https://beam.apache.org/documentation/sdks/javadoc/2.5.0/org/apache/beam/sdk/io/hdfs/HadoopFileSystemOptions.html">HadoopFileSystemOptions</a>
+   </td>
+   <td><a href="https://spark.apache.org/docs/latest/api/java/org/apache/spark/streaming/util/HdfsUtils.html">HdfsUtils</a>
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="2" >Object Stores
+   </td>
+   <td>Cloud Storage<br>(Using the <code>gs://</code> URI)
+   </td>
+   <td rowspan="2" ><a href="https://beam.apache.org/documentation/sdks/javadoc/2.5.0/org/apache/beam/sdk/io/hdfs/HadoopFileSystemOptions.html">HadoopFileSystemOptions</a>
```

Comment:
SImilar to above, maybe (GcsOptions or GcsFileSystem) and (S3Options or S3FileSystem) ?

Reply:
In this case, should it be: FileIO + GcsOptions, and FileIO + S3Options?

Reply:
Yes probably dividing both. I still don't understand why we refer to HadoopFileSystemOptions before instead of HadoopFileSystem?

Reply:
Somehow HadoopFileSystem doesn't seem to be documented. Is that intentional?

Reply:
Would you like to replace HadoopFileSystemOptions→HadoopFileSystem, GcsOptions→GcsFileSystem, and S3Options→S3FileSystem, and set the links to the source code in Github, since the *FileSystem classes aren't documented?